src/net.cpp

```diff
@@ -1406,6 +1421,12 @@ void CConnman::ThreadSocketHandler()
     }
 }
 
+void CConnman::SetMaxConnections(int newMaxConnections)
+{
+    nMaxConnections = std::min(newMaxConnections, nAvailableFds);
+}
```

Comment:
Maybe advertise the trigger of a cap (in both ways per above comment) here with:
```
if (nMaxConnections != newMaxConnections) {
    LogPrintf("%s: capped new maxconnections request of %d to %d", __func__, newMaxConnections, nMaxConnections);
}
```

Reply:
Done!